File: src/lib/jobs.ts
import { getRedis } from "@/lib/redis";

/** Enqueue a job for the agent worker onto the `jobs:agent` Redis Stream. */
export async function enqueue(type: string, payload: Record<string, unknown>): Promise<string> {
  const r = getRedis();
  const id = await r.xadd("jobs:agent", "*", "type", type, "payload", JSON.stringify(payload));
  return id ?? "";
}

File: worker_py/jobs.py
"""Worker-side job enqueue. Port of ``worker/lib/enqueue.ts`` / ``src/lib/jobs.ts``.

Lets agents kick off follow-up jobs (e.g. curator -> ingest_article) onto the
shared ``jobs:agent`` Redis Stream.
"""

import json

from redis.asyncio import Redis


async def enqueue(redis: Redis, job_type: str, payload: dict) -> str:
    job_id = await redis.xadd("jobs:agent", {"type": job_type, "payload": json.dumps(payload)})
    return job_id or ""
